mllib-local/src/main/scala/org/apache/spark/ml/linalg/BLAS.scala

```diff
@@ -267,7 +282,7 @@ private[spark] object BLAS extends Serializable {
       x: DenseVector,
       beta: Double,
       y: DenseVector): Unit = {
-    f2jBLAS.dspmv("U", n, alpha, A.values, x.values, 1, beta, y.values, 1)
+    javaBLAS.dspmv("U", n, alpha, A.values, x.values, 1, beta, y.values, 1)
```

Comment:
It looks like the intent here was to never try a vectorized implementation. That might or might not be wrong, but same point - I think we want to keep the separate native vs non-native options above, keep them separate

Reply:
The advantage of f2j over native is the absence of the overhead of switching between java and native. This overhead is particularly costly when the size of the inputs is small or whether the f2j implementation is just "good enough". The vector implementation is much faster than f2j on both small and large inputs (see https://github.com/luhenry/vectorizedblas/releases/tag/v0.1.11 -> jmh-results.csv for detailed numbers). It would then be wasteful not to take advantage of it when it's available.

Reply:
OK that's fine if we're fairly confident that this is a win for this use case. I don't even know if there was a strong reason for it or whether it's still valid.